tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py

```diff
@@ -0,0 +1,846 @@
+            warnings.warn(
+                f"Overriding config: out_block_h from {out_block_h_ntiles * 32} to user provided config={out_block_h_override}"
+            )
+            out_block_h_ntiles = out_block_h_override // 32
+    if "out_subblock_h" in config_override:
+        assert (
+            "out_subblock_w" in config_override
+        ), "out_subblock_w must also be provided as override config if out_subblock_h is provided"
+        out_subblock_h_override = config_override["out_subblock_h"]
+        assert out_subblock_h_override % 32 == 0, "out_subblock_h must be divisible by 32 (tile height)"
+        out_subblock_w_override = config_override["out_subblock_w"]
+        assert out_subblock_w_override % 32 == 0, "out_subblock_w must be divisible by 32 (tile width)"
+        if (out_subblock_h_override // 32) != out_subblock_h_ntiles:
+            warnings.warn(
+                f"Overriding config: out_subblock_h from {out_block_h_ntiles * 32} to user provided config={out_subblock_h_override}"
+            )
+        if (out_subblock_w_override // 32) != out_subblock_w_ntiles:
+            warnings.warn(
+                f"Overriding config: out_subblock_w from {out_subblock_w_ntiles * 32} to user provided config={out_subblock_w_override}"
+            )
+    if "out_subblock_w" in config_override:
+        assert (
+            "out_subblock_h" in config_override
+        ), "out_subblock_h must also be provided as override config if out_subblock_w is provided"
+
+    conv_blocking_config = ttl.tensor.OptimizedConvBlockConfig(
+        act_block_h_ntiles=act_block_h_ntiles,
+        act_block_w_ntiles=act_block_w_ntiles,
+        act_c_num_blocks=act_c_num_blocks,
+        weight_block_w_ntiles=weight_block_w_ntiles,
+        out_block_h_ntiles=out_block_h_ntiles,
+        out_subblock_h_ntiles=out_subblock_h_ntiles,
+        out_subblock_w_ntiles=out_subblock_w_ntiles,
+    )
+    return conv_blocking_config
+
+
+class TTPyCompositeConv(TTPyOp):
+    config_keys = [
+        "num_cores_nhw",
+        "grid_size",
+        "per_core_out_matrix_height",
+        "per_core_weight_matrix_width",
+        "act_block_h",
+        "act_block_w",
+        "act_c_num_blocks",
+        "weight_block_w",
+        "out_block_h",
+        "out_block_w",
+        "out_subblock_h",
+        "out_subblock_w",
+    ]
+
+    def __init__(
+        self,
+        sliding_window_op_params: Union[SlidingWindowOpParams, SlidingWindowOpParamsWithParallelConfig],
+        weight: ttl.tensor.Tensor,  # should user send TT tensor as weight tensor
+        output_channels,
+        input_channels,
+        device,
+        is_1d_systolic,
+        reader_patterns_cache,
+        bias: ttl.tensor.Tensor = None,
+        conv_blocking_and_parallelization_config_override=None,
+        fuse_relu=False,
+        weights_dtype=None,
+        output_dtype=None,
+        math_fidelity=None,
+        move_utwh_output=False,
```

Comment:
keep this signature same as the one in `ttnn`?

Reply:
Yes, will clean it up in another check-in. Ideally we would now use the TTNN op instead of this intermediate op directly in any model or test files.